main.go

```diff
@@ -93,7 +104,7 @@ func constructRecords(r resource.Resource) []string {
 	records = append(records, fmt.Sprintf("%s.%s.local. %d IN A %s", r.Name, r.Namespace, recordTTL, ip))
 	records = append(records, fmt.Sprintf("%s.in-addr.arpa. %d IN PTR %s.%s.local.", reverseIP, recordTTL, r.Name, r.Namespace))
 
-	if r.Namespace == defaultNamespace {
+	if r.Namespace == defaultNamespace || withoutNamespace {
```

Comment:
Would it make sense to _always_ advertise a record for ingress resources which honors the configured hostname on the Ingress object? The `without-namespaces` flag could then be used to omit the namespace if the resource type is a Service. For example:

```go
if r.Namespace == defaultNamespace || withoutNamespace || r.SourceType == "ingress" {
  # advertise records
}
```

I created a branch based off of your work which includes my suggestion. You can see the changeset here https://github.com/blake/external-mdns/commit/2693f40d26ddf90ea5fc568b188d35bcd03b34e1.

What do you think of this approach?

Reply:
Hey,

Sure, it makes sense.

Reply:
Do you wanna push here?

Reply:
If its alright with you, I can just merge this branch and squash these changes together into a single commit.